File: my-folder/0102-binary-tree-level-order-traversal/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<List<Integer>> list = new ArrayList<>();

    public List<List<Integer>> levelOrder(TreeNode root) {
        if(root == null){
            return list;
        }
        helper(root, 0);
        return list;
    }
    public void helper(TreeNode node, int level){
        if(list.size() == level){
            list.add(new ArrayList<>());
        }
        list.get(level).add(node.val);
        level++;
        if(node.left != null){
            helper(node.left, level);
        }
        if(node.right != null){
            helper(node.right, level);
        }
    }
}

File: my-folder/0102-binary-tree-level-order-traversal/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root: return []

        def bfs(root, arr):
            queue = [root]

            while queue:
                temp = []

                size = len(queue)
                
                for i in range(size):
                    node = queue.pop(0)
                    temp.append(node.val)
                    if node.left: queue.append(node.left)
                    if node.right: queue.append(node.right)

                arr.append(temp)

        res = []

        bfs(root, res)

        return res
